drivers/mbox/mbox_nrf_bellboard_local.c

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nordic_nrf_bellboard_local
+
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/mbox.h>
+#include <zephyr/sys/util.h>
+
+#include <hal/nrf_bellboard.h>
+
+#define BELLBOARD_NUM_IRQS 4U
+
+BUILD_ASSERT(DT_NUM_IRQS(DT_DRV_INST(0)) <= BELLBOARD_NUM_IRQS, "# interrupt exceeds maximum");
+
+BUILD_ASSERT(DT_INST_PROP_LEN(0, nordic_interrupt_mapping) == 2 * DT_NUM_IRQS(DT_DRV_INST(0)),
+	     "# interrupt mappings != # interrupts");
```

Comment:
This is also not needed. Just checking if the length is a multiple of 2 should be sufficient.